site/documentation/content/user-guide/mqtt-adapter.md

```diff
@@ -499,6 +507,139 @@ After a command has arrived as in the above example, the response is sent using
 mosquitto_pub -u 'gw@DEFAULT_TENANT' -P gw-secret -t command//4711/res/1010f8ab0b53-bd96-4d99-9d9c-56b868474a6a/200 -m '{"lumen": 200}'
 ```
 
+## Error Reporting via Error Topic
+
+The default behaviour when an error occurs while publishing telemetry, event or command response messages is for the MQTT adapter to close the network connection to the device, as mandated by the MQTT 3.1.1 spec.
+
+An alternative way of dealing with errors involves keeping the connection intact and letting the MQTT adapter publish a corresponding error message on a specific error topic to the device. To enable that behaviour, the device sends an MQTT *SUBSCRIBE* packet with a topic filter as described below on *the same MQTT connection* that is also used for publishing the telemetry, event or command response messages. Devices can subscribe with QoS 1 or QoS 0. The adapter indicates the outcome of the subscription request by sending back a corresponding *SUBACK* packet. The SUBACK packet will contain *Success - QoS 0* (`0x00`) or *Success - QoS 1* (`0x01`) for a valid error topic filter indicating QoS 0 or 1 and will contain the *Failure* (`0x80`) value for an invalid or unsupported filter. In order to again activate the default error handling behaviour, the device can send an MQTT *UNSUBSCRIBE* packet to the adapter, including the same topic filter that has been used to subscribe.
+
+The following sections define the topic filters to use for subscribing to error messages and the resulting error message topic. Instead of the `error` topic path segment, the shorthand version `e` is also supported.
+
+The following variables are used:
+
+* `${endpoint-type}`: The endpoint type of the device message that caused the error. Its value is either `telemetry`, `event` or the respective shorthand version. In case of a command response device message `command-response` or `c-s` is used.
+* `${correlation-id}`: The identifier that may be used to correlate the error message with the device message that caused the error. The identifier is either the value of a *correlation-id* property bag value contained in the device message topic, or the identifier is the *packet-id* of the device message if it was sent with QoS 1. Otherwise, a value of `-1` is used.
+* `${error-status}`: The HTTP status code of the error that was caused by the device message.
+
+{{% note title="Examples" %}}
+Since the subscription on the error topic needs to be done on the same MQTT connection that is also used for publishing the telemetry, event or command response messages, the Mosquitto MQTT Command Line Client cannot be used. The [MQTT CLI](https://hivemq.github.io/mqtt-cli/) tool with its [shell mode](https://hivemq.github.io/mqtt-cli/docs/shell.html) is an alternative that supports using one MQTT connection for both subscribing and publishing.
```

Comment:
BTW, I've tried using the `mosquitto_rr` client (mosquitto >= v1.6), but that doesn't work since mosquitto_rr already follows the MQTT 5 spec for Request/Response here (even for MQTT 3.11 servers) in that it doesn't allow wildcards in the response topic. Considering the error topic as a response topic wouldn't fit that well here anyway.